railties/test/railties/engine_test.rb

```diff
@@ -1508,18 +1512,32 @@ def index
       @plugin.write "app/controllers/bukkits/bukkit_controller.rb", <<-RUBY
         class Bukkits::BukkitController < ActionController::Base
           def index
-            render plain: main_app.bar_path
+            text = <<-TEXT
+              main_app.bar_path: \#{main_app.bar_path}
+              Rails.application.routes.url_helpers.bar_path: \#{Rails.application.routes.url_helpers.bar_path}
+            TEXT
+            render plain: text
           end
         end
       RUBY
 
       boot_rails
 
+      expected = <<-TEXT
+        main_app.bar_path: /foo/bar
+        Rails.application.routes.url_helpers.bar_path: /foo/bar
+      TEXT
       get("/bukkits/bukkit", {}, { "SCRIPT_NAME" => "/foo" })
-      assert_equal "/foo/bar", last_response.body
+      assert_equal expected.split("\n").map(&:strip),
+                   last_response.body.split("\n").map(&:strip)
```

Comment:
I think if we use `<<~TEXT` instead of `<<-TEXT` (for all strings), we can compare the strings directly:

```suggestion
      assert_equal expected, last_response.body
```